coding_systems/base/tests/helpers.py

```diff
@@ -0,0 +1,84 @@
+import pytest
+from django.test import TestCase
+
+
+class DynamicDatabaseStateNotRecordedError(Exception):
+    pass
+
+
+class DynamicDatabaseTestCase(TestCase):
+    """A special TestCase subclass to facilitate use of
+    a dynamically created database.
+
+    db_alias: name of the database that needs to be accessed, str (required)
+    coding_system: name of the coding system, for generating a temporary
+    database directory, str or None
+    import_data_path: name of the path to import data used, str or None
+    """
+
+    @property
+    def db_alias(self):
+        # The db_alias that will be added temporarily to the DB.
+        raise NotImplementedError(
+            "This test class requires a database alias to be set."
+        )
+
+    coding_system = None
+    import_data_path = None
+
+    @pytest.fixture
+    def _get_tmp_path(self, coding_systems_tmp_path):
+        """Runs the fixture that creates a temporary path."""
+        self.coding_systems_tmp_path = coding_systems_tmp_path
+        self.expected_db_path = (
+            self.coding_systems_tmp_path
+            / f"{self.coding_system}"
+            / f"{self.db_alias}.sqlite3"
+        )
+
+    @pytest.fixture(autouse=True)
+    def _configure_tmp_path(self, request):
+        if self.coding_system is not None:
+            request.getfixturevalue("_get_tmp_path")
```

Comment:
thought: This is slight magic to make the temporary path fixture an optional one. It may not be useful or necessary. It dynamically loads the `_get_tmp_path` fixture for tests that require it.

Reply:
What is the nature of the difference between those that do and don't want that behaviour?